tutorials/interoperability/cross-chain-dao.md

```diff
@@ -0,0 +1,884 @@
+
+Finally, the last thing to add is a vote mechanism that allows users to vote. This mechanism is nearly exactly the same as the mechanism in the [`GovernorCountingSimple` smart contract](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/governance/extensions/GovernorCountingSimple.sol){target=_blank}, so much of the code can be copied over:  
+
+```
+function castVote(uint256 proposalId, uint8 support) public virtual returns (uint256 balance)
+{
+    RemoteProposal storage proposal = proposals[proposalId];
+    require(
+        !proposal.voteFinished,
+        "DAOSatellite: vote not currently active"
+    );
+    require(
+        isProposal(proposalId), 
+        "DAOSatellite: not a started vote"
+    );
+
+    uint256 weight = token.getPastVotes(msg.sender, proposal.localVoteStart);
+    _countVote(proposalId, msg.sender, support, weight);
+
+    return weight;
+}
+
+function _countVote(uint256 proposalId, address account, uint8 support, uint256 weight) internal virtual 
+{
+    ProposalVote storage proposalVote = proposalVotes[proposalId];
+
+    require(!proposalVote.hasVoted[account], "DAOSatellite: vote already cast");
+    proposalVote.hasVoted[account] = true;
+
+    if (support == uint8(VoteType.Against)) {
+        proposalVote.againstVotes += weight;
+    } else if (support == uint8(VoteType.For)) {
+        proposalVote.forVotes += weight;
+    } else if (support == uint8(VoteType.Abstain)) {
+        proposalVote.abstainVotes += weight;
+    } else {
+        revert("DAOSatellite: invalid value for enum VoteType");
+    }
+}
+```
+
+Note that the `castVote` function requires that:  
+
+- The proposal isn't finished
+- The proposal exists, as in, there is data that's stored within the `proposals` map.
+
+In fact, the `_countVote` function is [directly copied](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/4fb6833e325658946c2185862b8e57e32f3683bc/contracts/governance/extensions/GovernorCountingSimple.sol#L78){target=_blank} from the hub chain! Much of the logic of single-chain dApps can be reused in cross-chain dApps with minor tweaks.  
+
+That's it for breaking down the satellite contract. It was more or less simple because most of the logic is just a reflection of what happens on the hub chain. You can view the completed smart contract in its [GitHub repository](https://github.com/jboetticher/cross-chain-dao/blob/main/contracts/DAOSatellite.sol){target=_blank}.  
+
+At this point, every single smart contract has been finished, and a deployment scheme like the one below can be made. If you are interested in seeing this in action, the [GitHub repository](https://github.com/jboetticher/cross-chain-dao){target=_blank} that hosts the cross-chain DAO allows you to deploy on TestNets.  
+
+![Smart contracts overview](/images/tutorials/interoperability/cross-chain-dao/cross-chain-dao-3.png)  
+
+## Caveats and Other Designs {: #caveats-and-other-designs }
```

Comment:
Maybe add here another disclaimer about the code not being meant for production but educational?